### PR TITLE
Work branch: Comments module

### PR DESCRIPTION
**Pull request by erastian**
- Add comments module.
- Fixed authorID type in ArticleCreateDTO
- Changed validation decorator for mongoDB ID fields in article-create.dto & comment-create.dto

### CODE REVIEW

File: src/articles/dto/article-create.dto.ts [hunk 1]
```diff
@@ -26,10 +26,10 @@ export class ArticleCreateDto implements Omit<Article, 'authorID' | 'id'> {
 	articleBody: string;
 
 	@IsEmpty()
-	authorID: number;
+	authorID: string;
 
 	@IsDefined()
-	@IsNumber()
+	@IsString()
```
Comment (mdziarkach) [inline src/articles/dto/article-create.dto.ts]: Add please custom decorator for cheching id
https://github.com/typestack/class-validator/issues/630

File: src/comments/comments.repository.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+						id: true,
+						username: true,
+						email: true,
+					},
+				},
+			},
+		}); //TODO: Add paginator. Better way get this query from article?
+	}
+
+	async findArticleComments(articleID: string, published: boolean | null): Promise<Comment[]> {
```
Comment (mdziarkach) [inline src/comments/comments.repository.ts]: create please 2 methods findPublishedCommentsByArticleId and findUnpublishedCommentsByArticleId

File: src/comments/comments.controller.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+			await this.commentsService.deleteComment(commentID);
+
+			this.ok(res, `Comment ${commentID} was successfully removed.`);
+		} catch (e) {
+			return next(e);
+		}
+	}
+
+	async findAllComments(req: Request, res: Response, next: NextFunction): Promise<void> {
```
Comment (mdziarkach) [inline src/comments/comments.controller.ts:68]: who will use it? I don't see any scenario when we need all comments

File: src/comments/dto/comment-create.dto.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { IsString, IsDefined, IsEmpty } from 'class-validator';
+
+export class CommentCreateDto implements Omit<Comment, 'authorID' | 'id'> {
+	@IsDefined()
+	@IsString()
+	commentBody: string;
+
+	@IsDefined()
+	@IsString()
+	articleID: string;
```
Comment (mdziarkach) [inline src/comments/dto/comment-create.dto.ts:11]: use custom middleware please for checking ids

File: src/comments/comments.controller.ts (hunk 3 above)
Comment (erastian) [reply src/comments/comments.controller.ts:68]: Moderator. One place for all comments with a lot of sorts and filters

File: src/articles/dto/article-create.dto.ts (hunk 1 above)
Comment (erastian) [reply src/articles/dto/article-create.dto.ts]: Done

File: src/comments/dto/comment-create.dto.ts (hunk 4 above)
Comment (erastian) [reply src/comments/dto/comment-create.dto.ts:11]: Done

File: src/comments/comments.controller.ts (hunk 3 above)
Comment (erastian) [reply src/comments/comments.controller.ts:68]: Decided to continue working on this endpoint with appropriate methods later

File: src/comments/comments.repository.ts (hunk 2 above)
Comment (erastian) [reply src/comments/comments.repository.ts]: Splited method in the comment repository module into 2 different methods for comments on articles for the roles of anonymous/user and moderator/admin

**Pull request merged**